### ISSUE TITLE
Add a view method to validate off-chain signatures

### ISSUE THREAD

**Issue by juniset** [OpenZeppelin/cairo-contracts#7]
On L1 the verification of off-chain signatures is (unfortunately...) typically achieved via `ecrecover`. This only works for EOA wallets. [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) was developed to verify the off-chain signatures of smart-contract wallets.

With Account Abstraction we need a generic interface to validate off-chain signatures and all accounts should implement that interface, i.e. we need to make sure it becomes a standard for Accounts. 

I would suggest something similar to EIP-1271:

```
const MAGIC_VALUE =  # get_selector_from_name('is_valid_signature')

@view
func is_valid_signature{
        storage_ptr: Storage*,
        pedersen_ptr: HashBuiltin*,
        ecdsa_ptr: SignatureBuiltin*,
        syscall_ptr: felt*,
        range_check_ptr
    } (
        hash: felt,
        signatures_len: felt,
        signatures: Signature*
    ) -> (magic_value: felt):
    # validate the signatures
    # return MAGIC_VALUE if the signatures are correct
end
```
where the method can receive an array of signatures to support EOA type accounts as well as multisig type accounts.

**Comment by frangio** [comment]
I agree that the account abstraction should contain an `is_valid_signature` function. Though I would say the interface should be agnostic to signing algorithm, including whether it is multisig or not. A pair `(signature_len: felt, signature: felt*)` should be enough, and a receiving contract could decode that into multiple signatures if it needs to. Does this sound like it would work?

Regarding the magic value, it sounds like it wouldn't be necessary in StarkNet because function selectors are 250 bits so collisions are basically impossible, unlike the 4 byte selectors in Ethereum which I think is the reason for using magic constants for return values there.
